sympy/series/limits.py

```diff
@@ -69,8 +70,17 @@ def heuristics(e, z, z0, dir):
         if r:
             rv = e.func(*r)
             if rv is S.NaN:
-                return
-
+                try:
+                    rat_e = ratsimp(e)
+                    if e != rat_e:
+                        e = rat_e if rat_e != S.NaN else e
+                        rv = limit(e, z, z0, dir)
```

Comment:
I was assuming if `rat_e` is `S.NaN` then it may lead to recursion error. So, I added the line 76. It basically checks if rat_e is nan or not . If it is then it calculates the limit using e. But, there is quite a possibility this may lead to a recursion error.